packages/firestore/src/remote/remote_store.ts

```diff
@@ -111,19 +108,25 @@ export class RemoteStore implements TargetMetadataProvider {
    */
   private listenTargets: { [targetId: number]: QueryData } = {};
 
-  private networkEnabled = false;
-
   private watchStream: PersistentListenStream;
   private writeStream: PersistentWriteStream;
   private watchChangeAggregator: WatchChangeAggregator = null;
 
   /**
    * Set to true by enableNetwork() and false by disableNetwork() and indicates
-   * the user-preferred network state. A network connection is only established
-   * if `networkAllowed` is true, the client is primary and there are
-   * outstanding mutations or active listens.
+   * the user-preferred network state.
+   */
+  private networkEnabled = true;
+
+  /**
+   * Indicates whether the network can accept requests (as determined by both
+   * the `isPrimary` flag and the user specified `networkEnabled` flag). If
+   * the connection is already established, new requests will be sent over the
+   * existing stream. If the stream has not yet been established, it will be
+   * established if there are outstanding requests.
    */
-  private networkAllowed = true;
+  private canUseNetwork = false;
```

Comment:
Hrm.... My hope would be that we don't need to track 3 booleans and instead canUseNetwork would instead be a function of networkEnabled and isPrimary.  As-is I'm not sure what they're relation is.

Reply:
We need a boolean to indicate whether the network connection is established, so that we can shut it down when needed in `disableNetwork()`. At first, this was called `networkEnabledInternal`, but I then realizes that I need the same state in the `canAddToWritePipeline` and in the asserts that verify that a stream didn't close unexpectedly. 

If I want to use a separate `canUseNetwork()` method that uses both isPrimary and networkEnabled, then I need to flip networkEnabled to false during shutdown. Otherwise, I am hitting a bunch of asserts in the teardown.